change interface to add support of psr/http-message:2.0

AleksandrCherepanov opened the pull request.

AleksandrCherepanov committed:
change interface to add support of psr/http-message:2.0

arokettu commented on composer.json:
```diff
@@ -38,7 +38,7 @@
         "ext-libxml": "*",
         "psr/http-factory": "^1.0",
-        "psr/http-message": "^1.1"
+        "psr/http-message": "^2.0"
```
Tests seem to pass with 1.1 too, maybe relax to "^1.1 | ^2.0"?

AleksandrCherepanov replied:
Hi, sorry for the delay. Seems reasonable, added

AleksandrCherepanov committed:
make psr/http-message less string with 1.1 and 2.0 versions

akrabat committed:
Update http-message constraint

akrabat committed:
Fix Ci for PHP 8.0

The composer.lock doesn't work for PHP 8.0, so rather than use the lock
file, do a composer update.

akrabat commented:
Thanks.

Donov4n commented:
Hello,

These changes prevent proper `Response` method chaining without triggering errors from static analysis tools...

I can no longer do:

```php
return $response
    ->withStatus(StatusCode::STATUS_OK)
    ->withFile($myFile);
```

because with this PR the return type of withStatus is not `: static` but `Psr\Http\Message\ResponseInterface`, and since `withFile()` is not part of that interface (but is defined in the `Slim\Http\Response` decorator), we get `Undefined method 'withFile'.` error.

Wouldn’t returning `: static` instead of  (PSR-7) `: ResponseInterface` have been enough to remain compliant?
